accounts/abi/pack_test.go

```diff
@@ -322,7 +329,41 @@ func TestPack(t *testing.T) {
 		{
 			"string",
 			"foobar",
-			common.Hex2Bytes("0000000000000000000000000000000000000000000000000000000000000006666f6f6261720000000000000000000000000000000000000000000000000000"),
+			common.Hex2Bytes("0000000000000000000000000000000000000000000000000000000000000020" + // offset: 32
+				"0000000000000000000000000000000000000000000000000000000000000006" + // len: 6
+				"666f6f6261720000000000000000000000000000000000000000000000000000"), // "foobar"
+		},
+		{
+			"string[]",
+			[]string{"hello", "foobar"},
+			common.Hex2Bytes("0000000000000000000000000000000000000000000000000000000000000020" + // offset array 32
+				"0000000000000000000000000000000000000000000000000000000000000002" + // len(array) = 2
+				"0000000000000000000000000000000000000000000000000000000000000040" + // offset 64 to i=0
+				"0000000000000000000000000000000000000000000000000000000000000080" + // offset 128 to i=1
+				"0000000000000000000000000000000000000000000000000000000000000005" + // len(str[0]) = 5
+				"68656c6c6f000000000000000000000000000000000000000000000000000000" + // str[0]
+				"0000000000000000000000000000000000000000000000000000000000000006" + // len(str[1]) = 6
+				"666f6f6261720000000000000000000000000000000000000000000000000000"), // str[1]
+		},
+		{
+
+			//web3.eth.abi.encodeParameter('bytes32[][]', [['0x0100000000000000000000000000000000000000000000000000000000000000',
```

Comment:
I added the web3.js calls as a comment for easier cross checking. We can remove them later